### PR TITLE
Add ESNet to Matbench Discovery leaderboard

### PR DESCRIPTION
**Pull request by zzz-sl**
## Checklist

Please check the following items before submitting your PR:

- [√ ] I created a new folder and YAML metadata file `models/esnet/esnet.yml` for my submission. `esnet` is the name of the architecture and `esnet.yml` includes things like author details, training set names and important hyperparameters.
- [√ ] I added the my new model as a new attribute on the [`Model.esnet` enum](https://github.com/janosh/matbench-discovery/blob/57d0d0c8a14cd317/matbench_discovery/enums.py#L274) in `enums.py`.
- [ √] I uploaded the energy model prediction file for the WBM test set to Figshare or another cloud storage service (`[2025-05-08-esnet-wbm-IS2RE.csv.gz](https://figshare.com/ndownloader/files/54445643)`).
- [√ ] I included the urls to the Figshare files in the YAML metadata file (`models/esnet/esnet.yml`). If not using Figshare I have included the urls to the cloud storage service in the description of the PR.
- [√ ] I included the test script (`test_esnet_discovery.py` for `task` in `discovery`, `kappa`, `diatomics`) that generated the prediction files.

## Additional Information (Optional)
Tisp:
-  Prediction result url:  https://zenodo.org/records/15387024 or https://figshare.com/files/54445643
-  checkpoints_url: https://figshare.com/files/54259907
-  ESNet model repository: https://github.com/zzz-sl/ESNet

### CODE REVIEW

**Comment by janosh** [comment]
thanks for this submission 👍 

there's something wrong with `models/esnet/2025-05-08-esnet-wbm-IS2RE.csv.gz`, the column `e_form_per_atom_esnet` doesn't seem to contain formation energies:


e-form-parity-esnet

**Comment by zzz-sl** [comment]
Thank you for flagging this issue! We will verify the problem. As part of our ongoing improvements, we are retraining the model using the MP2022 dataset and will resubmit updated weights and corrected predictions once the training and validation are complete. We appreciate your feedback.

**Comment by zzz-sl** [comment]
Update prediction result url:  https://figshare.com/files/55207640  and checkpoint_url:  https://figshare.com/files/55207664

**Comment by zzz-sl** [comment]
Add the test set metric values in esnet.yaml, and simultaneously update the pred_file_url=https://figshare.com/files/55539980 and checkpoint_url=https://figshare.com/files/55540055

**Review by janosh — APPROVED** [review]
all looks good! thanks a lot for this submission 👍 and apologies for the delay in reviewing.
